java-security/src/test/java/com/sap/cloud/security/token/validation/validators/SapIdJwtSignatureValidatorTest.java

```diff
@@ -170,6 +170,21 @@ public void validate_withEnabledProofTokenCheck() throws IOException {
 		assertTrue(cut.validate(iasToken).isValid());
 	}
 
+	@Test
+	public void validate_withEnabledProofTokenCheck_app2app() {
```

Comment:
Does this test really test the new functionality? I think we must check if tokenKeyServiceMock#retrieveTokenKeys has been called with or without certificate.